doc, gen_vimdoc.py

justinmk opened the pull request:
`scripts/gen_vimdoc.py` (renamed from `gen_vimdoc.py`) now supports nested elements and markdown-formatted lists. 

Note: Doxygen does a pretty nice job of interpreting the quasi-Markdown in our doctsrings, but it can't make sense when we do things like this:

```
- foo1
  - foo2
zub
- bar1
```

The guideline is: if you want indented things, they must be lists. That `zub` in the above tree is parsed by Doxygen as belonging to the `foo2` element. Adding line breaks causes different problems... Easiest solution is for the docstring author to make `zub` a list item (`- zub`) with `- bar1` is nested in it.

todo

- [x] https://github.com/neovim/neovim/pull/9710

justinmk committed:
doc [ci skip]

closes #9719

bfredl commented on src/nvim/api/buffer.c:
```diff
@@ -972,7 +972,7 @@ Integer nvim_buf_add_highlight(Buffer buffer,
 /// line_start and line_end respectively.
 ///
-/// @param buffer     Buffer handle
+/// @param buffer     Buffer handle, or 0 for current buffer
```
Why change this on one function only?

justinmk replied:
I'll change it for all cases. I was trying to use this one for a script and noticed it wasn't specified.

justinmk committed:
gen_vimdoc.py: render nested lists, etc [ci skip]

- render_node() is now the main rendering function: it traverses a node
  and builds the Vim help text recursively.
- render_para() is weird and ugly, it is the entry-point for rendering
  the help text for one docstring'd function.

mhinz commented:
https://github.com/neovim/neovim/pull/9722#pullrequestreview-213515458

bfredl commented:
@mhinz no need to change that,`winhl=Normal:XXX` works perfectly well for any window. (No need to make floats seem needing special treatment when they don't )

justinmk committed:
doc: move ui-wildmenu to deprecated.txt [ci skip]

justinmk commented on runtime/doc/deprecated.txt:
```diff
@@ -63,6 +63,15 @@ Options ~
+			backwards-compatibility:
+				["wildmenu_show", items]
+				["wildmenu_select", selected]
+				["wildmenu_hide"]
```
@bfredl I moved this to deprecated.txt.

justinmk committed:
doc: mention "pynvim" module rename

closes #9764

justinmk commented on src/nvim/api/vim.c:
```diff
@@ -1306,14 +1309,14 @@ Array nvim_get_api_info(uint64_t channel_id)
-///                 "GPLv3" or "MIT"
-///     - "logo"    URI or path to image, preferably small logo or icon.
-///                 .png or .svg format is preferred.
+/// @param attributes Arbitrary string:string map of informal client properties.
```
Restricting this to string-only values. https://github.com/neovim/go-client/pull/36#issuecomment-476418658

bfredl commented on runtime/doc/map.txt:
```diff
@@ -274,27 +274,31 @@ Using 0x80 as a single byte before other text does not work, it will be seen
+Note:
+- Because <Cmd> avoids mode-changes (unlike ":") it does not trigger
+  |CmdlineEnter| and |CmdlineLeave| events. This helps performance.
+- For the same reason, |keycodes| like <C-R><C-W> are interpreted in the
```
@justinmk I don't think this is correct, try

    noremap <f2> <cmd>echo "<c-r><c-w>"<cr>
    noremap! <f2> <cmd>echo "<c-r><c-w>"<cr>

and press `<f2>` in various modes. The result is always the same, `<c-r><c-w>` are always interpreted as the bytes they represent, not according to any mode.

justinmk replied:
127e13f53ee0
